File: BeagleBone/Black/gpiod/get.c
// //////////////////////////////////////
// 	get.c
//  Get the value of P8_13. P8_13 is line 23 on chip 0.
// 	Wiring:	Attach a switch to P8_13 and 3.3V
// 	Setup:	sudo apt uupdate; sudo apt install libgpiod-dev
// 	See:	https://github.com/starnight/libgpiod-example/blob/master/libgpiod-led/main.c
// //////////////////////////////////////
#include <gpiod.h>
#include <stdio.h>
#include <unistd.h>

#define	CONSUMER	"Consumer"

int main(int argc, char **argv)
{
	int chipnumber = 0;
	unsigned int line_num = 23;	// GPIO Pin P8_13
	unsigned int val;
	struct gpiod_chip *chip;
	struct gpiod_line *line;
	int i, ret;

	chip = gpiod_chip_open_by_number(chipnumber);
	if (!chip) {
		perror("Open chip failed\n");
		goto end;
	}

	line = gpiod_chip_get_line(chip, line_num);
	if (!line) {
		perror("Get line failed\n");
		goto close_chip;
	}

	ret = gpiod_line_request_input(line, CONSUMER);
	if (ret < 0) {
		perror("Request line as output failed\n");
		goto release_line;
	}

	/* Get */
	while(1) {
		printf("%d\r", gpiod_line_get_value(line));
		usleep(1000);
	}

release_line:
	gpiod_line_release(line);
close_chip:
	gpiod_chip_close(chip);
end:
	return 0;
}

File: BeagleBone/Black/gpiod/get.py
#!/usr/bin/env python3
# //////////////////////////////////////
# 	get.py
#  Get the value of P8_13. P8_13 is line 23 on chip 0.
# 	Wiring:	Attach a switch to P8_13 and 3.3V
# 	Setup:	sudo apt uupdate; sudo apt install libgpiod-dev
# 	See:	https://github.com/starnight/libgpiod-example/blob/master/libgpiod-led/main.c
# //////////////////////////////////////

import gpiod
import sys

if len(sys.argv) < 3:   # Use P8_13 if not given
    CHIP='0'
    offsets=[23]
else:
    CHIP=sys.argv[1]
    offsets = []
    for off in sys.argv[2:]:
        offsets.append(int(off))
    
chip = gpiod.Chip(CHIP)

lines = chip.get_lines(offsets)
lines.request(consumer=sys.argv[0], type=gpiod.LINE_REQ_DIR_IN)

print("Hit ^C to stop")

while True:
    vals = lines.get_values()
    
    for val in vals:
        print(val, end=' ')
    print('\r', end='')
